packages/system/src/hooks.ts

```diff
@@ -19,14 +19,33 @@ export function useChakra<T extends Dict = Dict>() {
   return { ...colorModeResult, theme }
 }
 
+const resolveBreakpointValue = <T extends StringOrNumber>(
+  theme: Dict,
+  tokenValue: T,
+  fallbackValue: any,
+) => {
+  if (tokenValue === null) return tokenValue
+
+  const getValue =
+    typeof tokenValue === "number"
+      ? (val: T) => theme.__breakpoints?.asArray?.[val]
+      : (val: T) => theme.__breakpoints?.asObject?.[val]
+  return getValue(tokenValue) ?? getValue(fallbackValue) ?? fallbackValue
+}
+
 // inspired from ./css.ts : resolveTokenValue
 const resolveTokenValue = <T extends StringOrNumber>(
   theme: Dict,
   tokenValue: T,
   fallbackValue: any,
 ) => {
   if (tokenValue == null) return tokenValue
-  const getValue = (val: T) => theme.__cssMap?.[val]?.value
+  const scale = tokenValue.toString().split(".")[0]
+  const getValue =
+    scale !== "breakpoints"
+      ? (val: T) => theme.__cssMap?.[val]?.value
+      : (val: T) => theme.__breakpoints?.[val]?.value
+  // const getValue = (val: T) => theme.__cssMap?.[val]?.value
```

Comment:
The code here feels a bit more complex than needed. Maybe we can collaborate on this to find opportunities to simplify this.